rosidl_generator_c/CMakeLists.txt

```diff
@@ -2,44 +2,12 @@ cmake_minimum_required(VERSION 3.5)
 
 project(rosidl_generator_c C)
 
-# Default to C11
-if(NOT CMAKE_C_STANDARD)
-  set(CMAKE_C_STANDARD 11)
-endif()
-
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic)
-endif()
```

Comment:
For the remaining tests we still want to keep the C standard and compiler flags I would assume.

Reply:
Restored https://github.com/ros2/rosidl/pull/442/commits/e3fe8789ba6237adb9da5e347e0ed90b3c1a7448